### DISCUSSION TITLE
Questions about K2ForJ4

### DISCUSSION THREAD

**Discussion by cmirnow** [eworkers/K2ForJ4#49] [General]
I'll ask those questions here. Because they're not bugs, they're questions.

1. Is it possible to update from Joomla 4 with K2ForJ4 to Joomla 5?
2. After upgrading from Joomla 3 to Joomla 4 - com_k2_extended cache size increases two or three times in standard caching. What do you think about this, is this a consequence of the problem or is it ok?
3. If an official version of K2 for Joomla 4 is published, will it be possible to upgrade from K2ForJ4 to K2?

**Comment by eworkers** [comment]
> I'll ask those questions here. Because they're not bugs, they're questions.

> 1. Is it possible to update from Joomla 4 with K2ForJ4 to Joomla 5?


**Yes**


> 2. After upgrading from Joomla 3 to Joomla 4 - com_k2_extended cache size increases two or three times in standard caching. What do you think about this, is this a consequence of the problem or is it ok?

Thanks to you, this is not the case anymore (During the development process, J4 was at RC stage and many APIs have been changed since, we faced some serialization issues that apparently were linked to Joomla itself, so we introduced a workaround for some caching issues that hopefully are not required anymore)

> 3. If an official version of K2 for Joomla 4 is published, will it be possible to upgrade from K2ForJ4 to K2?

**Yes**, what made our work so hassling was the fact to not break future compatibility with potential future K2 releases (No db change, No API change), more over the current fork still point to K2 official update link and we did not increase the version number in our code. This means that whenever an official k2 version is released you will be notified to update. Just make sure by then that the official K2 release is compatible with J4 or J5 and not another <=J3 compatible release.

**Comment by cmirnow** [comment]
You're right, I do keep a close eye on caching. My blog is on AWS and I have to strictly monitor resource consumption.
:D
That was a joke. Thank you!

**Comment by litho-publisher** [comment]
after chance to joomla 4 and than to 5 I get this error bei using the User Menu as frontend modul. What can I do?

                  ERROR 0 - Class "JSite" not found in 
..../htdocs/modules/mod_k2_user/helper.php:110


# | Function | Location
-- | -- | --
1 | () | JROOT/modules/mod_k2_user/helper.php:110
2 | modK2UserHelper::getMenu() | JROOT/modules/mod_k2_user/mod_k2_user.php:83
3 | include() | JROOT/libraries/src/Dispatcher/ModuleDispatcher.php:52
4 | Joomla\CMS\Dispatcher\ModuleDispatcher::Joomla\CMS\Dispatcher\{closure}() | JROOT/libraries/src/Dispatcher/ModuleDispatcher.php:55
5 | Joomla\CMS\Dispatcher\ModuleDispatcher->dispatch() | JROOT/libraries/src/Helper/ModuleHelper.php:291
6 | Joomla\CMS\Helper\ModuleHelper::renderRawModule() | JROOT/libraries/src/Helper/ModuleHelper.php:162
7 | Joomla\CMS\Helper\ModuleHelper::renderModule() | JROOT/libraries/src/Document/Renderer/Html/ModuleRenderer.php:99
8 | Joomla\CMS\Document\Renderer\Html\ModuleRenderer->render() | JROOT/libraries/src/Document/Renderer/Html/ModulesRenderer.php:51
9 | Joomla\CMS\Document\Renderer\Html\ModulesRenderer->render() | JROOT/libraries/src/Document/HtmlDocument.php:575
10 | Joomla\CMS\Document\HtmlDocument->getBuffer() | JROOT/libraries/src/Document/HtmlDocument.php:894
11 | Joomla\CMS\Document\HtmlDocument->_renderTemplate() | JROOT/libraries/src/Document/HtmlDocument.php:647
12 | Joomla\CMS\Document\HtmlDocument->render() | JROOT/libraries/src/Application/CMSApplication.php:1073
13 | Joomla\CMS\Application\CMSApplication->render() | JROOT/libraries/src/Application/SiteApplication.php:732
14 | Joomla\CMS\Application\SiteApplication->render() | JROOT/libraries/src/Application/CMSApplication.php:311
15 | Joomla\CMS\Application\CMSApplication->execute() | JROOT/includes/app.php:58
16 | require_once() | JROOT/index.php:32

</div></div>
</body>
</html>

**Reply by eworkers** [reply]
please don't report an issue in discussion boards. Your issue should be closed after last push
